internal/compiler/parser.rs

```diff
@@ -322,7 +322,7 @@ declare_syntax! {
     }
     // syntax kind
     {
-        Document -> [ *Component, *ExportsList, *ImportSpecifier, *StructDeclaration ],
+        Document -> [ *Component, *ExportsList, *ImportSpecifier, *StructDeclaration, *AtPragma ],
```

Comment:
It seems to me that this is the wrong location in the "AST". I think it should be attached (as child) to those elements where it applies, i.e. initially just structure declarations.

Reply:
But as Struct is Document element , I suppose AtPragma according to its syntax it should be an specif Document element which contains Struct . isn't it ?